packages/shared/src/hooks/integrations/useIntegrationQuery.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  useQuery,
+  UseQueryOptions,
+  UseQueryResult,
+} from '@tanstack/react-query';
+import { UserIntegration } from '../../graphql/integrations';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { gqlClient } from '../../graphql/common';
+import { USER_INTEGRATION_BY_ID } from '../../graphql/users';
+import { useAuthContext } from '../../contexts/AuthContext';
+
+export type UseIntegrationQueryProps = {
+  id?: string;
+  queryOptions?: UseQueryOptions<UserIntegration>;
+};
+
+export type UseIntegrationQuery = UseQueryResult<UserIntegration>;
+
+export const useIntegrationQuery = ({
+  id,
+  queryOptions,
```

Comment:
To avoid breaking the app.

```suggestion
  queryOptions = {},
```